pennylane_qiskit/qiskit_device2.py

```diff
@@ -98,7 +98,31 @@ def circuit(x):
     """
     # Code to acquire session:
     existing_session = device._session
-    session = Session(backend=device.backend)
+    session_options = {
+        "service": device.service,
+        "backend": device.backend,
+    }
+
+    for k, v in kwargs.items():
+        # Options like service and backend should be tied to the settings set on device
+        if k in session_options and k != "max_time":
+            warnings.warn(f"Using '{k}' set in device, {device.backend}", UserWarning)
+
+        # Need "_" since `max_time` attribute on Session is `_max_time`
+        # When there is overlap between the Session options on the device and the session options
+        # passed in via qiskit_session, we prefer the ones passed in via qiskit_session
+        elif existing_session and (
+            hasattr(existing_session, "_" + k) or hasattr(existing_session, k)
+        ):
+            warnings.warn(
+                f"`{k}` was set in the Session passed to the device. Using `{k}` '{v}' set in `qiskit_session`.",
+                UserWarning,
+            )
+            session_options[k] = v
+        else:
+            session_options[k] = v
```

Comment:
If Qiskit releases new Session options, this code would still capture those options and raise appropriate warnings if overlapped (unless they make one of the session attributes have extremely weird naming convention)